Shared/ViewModels/LatestInLibraryViewModel.swift

```diff
@@ -0,0 +1,75 @@
+//
+// Swiftfin is subject to the terms of the Mozilla Public
+// License, v2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at https://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) 2023 Jellyfin & Jellyfin Contributors
+//
+
+import Combine
+import Foundation
+import JellyfinAPI
+
+final class LatestInLibraryViewModel: PagingLibraryViewModel {
+
+    let parent: LibraryParent
+
+    init(parent: LibraryParent) {
+        self.parent = parent
+
+        super.init()
+
+        _requestNextPage()
+    }
+
+    override func _requestNextPage() {
+        Task {
+
+            await MainActor.run {
+                self.isLoading = true
+            }
+
+            let parameters = Paths.GetLatestMediaParameters(
+                parentID: self.parent.id,
+                fields: ItemFields.minimumCases,
+                enableUserData: true,
+                limit: self.pageItemSize * 3
+            )
+            let request = Paths.getLatestMedia(userID: userSession.user.id, parameters: parameters)
+            let response = try await userSession.client.send(request)
+
+            let items = response.value
+            if items.isEmpty {
+                hasNextPage = false
+                return
+            }
+
+            await MainActor.run {
+                self.isLoading = false
+                self.items.append(contentsOf: items)
+            }
+        }
+    }
+
+    override public func getRandomItemFromLibrary() async throws -> BaseItemDtoQueryResult {
+        if items.isEmpty {
```

Comment:
Sadly I let this function pass as is, since it should just return the item, but this just needs:
```swift
BaseItemDtoQueryResult(items: items.elements)
```

since right now the button handles choosing a random item. I will change that.

Reply:
Done...

Reply:
Looks like you still need to lint.

Reply:
Done